plugins/database/src/db_plugin.cpp

```diff
@@ -14529,13 +14529,283 @@ irods::error db_get_hierarchy_for_resc_op(
 
 } // db_get_hierarchy_for_resc_op
 
+namespace
+{
+    // A support function for executing SQL operations.
+    irods::error execute_sql(const std::string_view _sql)
+    {
+        auto ec = cmlExecuteNoAnswerSql(_sql.data(), &icss);
+        if (CAT_SUCCESS_BUT_WITH_NO_INFO == ec) {
+            return CODE(ec);
+        }
+
+        if (0 != ec) {
+            logger::database::error("SQL execution error [{}].", ec);
+            return ERROR(ec, "SQL execution error.");
+        }
+
+        ec = cmlExecuteNoAnswerSql("commit", &icss);
+        if (ec < 0) {
+            return ERROR(ec, "Failed to commit SQL updates.");
+        }
+
+        return SUCCESS();
+    } // execute_sql
+
+    irods::error process_ticket_operation_as_admin(irods::plugin_context& _ctx,
```

Comment:
What do you think of the name `execute_ticket_operation_as_admin`?

Reply:
That sounds better. Will update.

Reply:
i like the new name better, but let's change this one after this PR has been merged. i feel i'll forget to update this in the main-line branch.

Reply:
i changed my mind again. im going to change this in both branches lol